nil/internal/execution/block_generator.go

```diff
@@ -294,6 +301,16 @@ func (g *BlockGenerator) handleTxn(txn *types.Transaction) error {
 	g.addReceipt(res)
 	g.counters.CoinsUsed = g.counters.CoinsUsed.Add(res.CoinsUsed())
 
+	if assert.Enable && txn.IsExternal() {
+		newSeqno, err := g.executionState.GetExtSeqno(txn.To)
+		check.PanicIfErr(err)
+		if res.GasUsed == 0 {
+			check.PanicIfNotf(newSeqno == seqno, "seqno changed during execution")
+		} else {
+			check.PanicIfNotf(newSeqno == seqno+1,
+				"seqno was not changed during execution (old %d, new: %d). Gas used: %d", seqno, newSeqno, res.GasUsed)
```

Comment:
Seems like a bad message
If `newSeqno` is equal to `seqno+2`, then the message "seqno was not changed during execution" will be wrong. And if `newSeqno == seqno`, then the `(old %d, new: %d)` part will have two identical values
I would reword)

Reply:
Ok. Changed a bit. Now it's "...seqno was not changed **correctly**..."